de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/celleditor/DefaultVerificationTypeReferenceCellEditingSupport.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.extension.requirements.ui.celleditor;
+
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.jface.viewers.ColumnViewer;
+
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.ReferenceProperty;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ComposedPropertyInstance;
+import de.dlr.sc.virsat.model.extension.requirements.model.DefaultVerification;
+import de.dlr.sc.virsat.model.extension.requirements.model.VerificationType;
+import de.dlr.sc.virsat.uiengine.ui.cellEditor.aproperties.ReferencePropertyCellEditingSupport;
+
+/**
+ * Class simply using bean setters for default verification type
+ *
+ */
+public class DefaultVerificationTypeReferenceCellEditingSupport extends ReferencePropertyCellEditingSupport {
+
+	/**
+	 * @param editingDomain
+	 * @param viewer
+	 * @param property
+	 */
+	public DefaultVerificationTypeReferenceCellEditingSupport(EditingDomain editingDomain, ColumnViewer viewer,
+			ReferenceProperty property) {
+		super(editingDomain, viewer, property);
+	}
+
+	@Override
+	protected Command createSetCommand(Object element, Object userInputValue) {
+		CategoryAssignment caDefaultVerification = null;
+		CategoryAssignment caType = null;
+		if (element instanceof CategoryAssignment) {
+			caDefaultVerification = (CategoryAssignment) element;
+		} else if (element instanceof ComposedPropertyInstance) {
+			caDefaultVerification = ((ComposedPropertyInstance) element).getTypeInstance();
+		} else {
+			return super.createSetCommand(element, userInputValue);
+		}
+		if (userInputValue instanceof CategoryAssignment) {
+			caType = (CategoryAssignment) userInputValue;
+		}
+		if (caDefaultVerification.getType().getFullQualifiedName().equals(DefaultVerification.FULL_QUALIFIED_CATEGORY_NAME)

```

Comment:
There seems to be quite a bit of if branching going on in this method.
Could we maybe move this if block into the above if?
Then we could also move the caType declaration inside and could drop the `caType != null` in the if.

Reply:
Not really sure if it's better like this, because we now still have to check if caType is a CA and then also check if it is the right CA... all in one IF... but it's less branching now :D

Reply:
Hmm yeah, it looked nicer in my head.
In the future we should maybe consider making a helper function in CategoryAssignmentHelper to check if an object is a ca & has the correct type given by a fqn.
I believe we probably have to check both in a lot of places.